ui-performance/index.md

```diff
@@ -121,27 +121,51 @@ Gallery example:
 <center>The graphs produced by the performance overlay.<br>The vertical green
 bars represent the current frame.</center><br>
 
-Each graph represents the last 300 frames for the following two threads used
-in every Flutter app:
+Flutter uses several threads to do its work. You have direct control
+over what happens in the UI thread, but no direct access to the other
+threads.
+
+1. UI thread<br>
+   The platform's main thread executes Dart code in the Dart VM.
+   This thread includes code that you wrote, and code executed by
+   Flutter's framework on your app's behalf.
+   Also referred to as Flutter's _engine_, plugins typically run here.
+
+1. Dart application thread<br>
+   [PENDING: Not sure what runs here, since I thought all Dart code
+   ran in the UI thread...]
+
+1. GPU or rasterizer thread<br>
+   The thread responsible for painting your application, or
+   interfacing with the phone's GPU (graphic processing unit) thread.
+   When your app creates and displays a scene, the framwork, creates a
+   "layer tree", a lightweight object containing device-agnostic
+   painting commands, and sends the layer tree to the GPU thread to
+   be rendered on the device. You cannot directly access the GPU thread or
+   its data but, if this thread is slow, it's a result of something
+   you've done in the Dart code.
+   Skia, the graphics library, runs on this thread, which also
+   handles image I/O. [PENDING: Does it handle image I/O?]
+
+1. I/O thread<br>
```

Comment:
I feel like this whole section section can be summarized as follows:

1: Platform Thread: The platforms main thread. Plugin code runs here. (You may add links to UIKit and Android documentation about what they consider their main thread).
2: UI Thread: Runs your Dart code. Generate the layer tree. Don't block this!
3: GPU Thread: Takes layer tree and displays it by talking to the GPU.
4: IO Thread: Performs expensive tasks (mostly IO) that would otherwise block either the UI or GPU threads.

Reply:
Done